File: test/data/cpp/simple.cpp
// Simple C++ test file

#include <iostream>
#include <vector>
#include <memory>

namespace utils {
    template<typename T>
    class Container {
    public:
        Container() : size_(0) {}
        
        void add(const T& item) {
            items_.push_back(item);
            size_++;
        }
        
        T get(size_t index) const {
            return items_[index];
        }
        
        size_t size() const { return size_; }
        
    private:
        std::vector<T> items_;
        size_t size_;
    };
}

class Calculator {
public:
    Calculator() : result_(0) {}
    
    int add(int a, int b) {
        result_ = a + b;
        return result_;
    }
    
    int multiply(int x, int y) {
        return x * y;
    }
    
    // Operator overloading
    Calculator& operator+=(int value) {
        result_ += value;
        return *this;
    }
    
private:
    int result_;
};

// Function template
template<typename T>
T max(T a, T b) {
    return (a > b) ? a : b;
}

// Free function
void printMessage(const std::string& msg) {
    std::cout << "Message: " << msg << std::endl;
}

int main() {
    // Using the Calculator class
    Calculator calc;
    int sum = calc.add(5, 3);
    
    // Using the template class
    utils::Container<int> numbers;
    numbers.add(10);
    numbers.add(20);
    
    // Using function template
    int maxInt = max(10, 20);
    double maxDouble = max(3.14, 2.71);
    
    // Lambda expression
    auto multiply = [](int a, int b) -> int {
        return a * b;
    };
    
    int product = multiply(4, 5);
    
    return 0;
}

File: test/data/csharp/simple.cs
// Simple C# example demonstrating modern language features
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SittingDuck.Examples
{
    // Interface definition
    public interface IGreeter
    {
        string Greet(string name);
    }

    // Record type (C# 9+)
    public record Person(string Name, int Age);

    // Main class with various features
    public class Calculator : IGreeter
    {
        // Properties
        public int Result { get; private set; }

        // Auto-implemented property with initializer
        public string Version { get; } = "1.0.0";

        // Constructor
        public Calculator()
        {
            Result = 0;
        }

        // Method with expression body
        public int Add(int a, int b) => a + b;

        // Traditional method
        public int Subtract(int a, int b)
        {
            return a - b;
        }

        // Async method
        public async Task<string> FetchDataAsync()
        {
            await Task.Delay(100);
            return "Data fetched";
        }

        // Interface implementation
        public string Greet(string name)
        {
            return $"Hello, {name}!";
        }

        // Static method
        public static Calculator Create() => new Calculator();

        // Method with pattern matching
        public string Classify(object obj)
        {
            return obj switch
            {
                int i when i > 0 => "Positive integer",
                int i when i < 0 => "Negative integer",
                string s => $"String: {s}",
                null => "Null value",
                _ => "Unknown type"
            };
        }
    }

    // Enum definition
    public enum Status
    {
        Active,
        Inactive,
        Pending
    }

    // Struct definition
    public struct Point
    {
        public int X { get; set; }
        public int Y { get; set; }

        public Point(int x, int y)
        {
            X = x;
            Y = y;
        }

        public double Distance() => Math.Sqrt(X * X + Y * Y);
    }

    // Generic class
    public class Container<T>
    {
        private readonly List<T> _items = new();

        public void Add(T item) => _items.Add(item);

        public T? GetFirst() => _items.FirstOrDefault();

        public IEnumerable<T> GetAll() => _items;
    }

    // Program entry point
    public class Program
    {
        public static void Main(string[] args)
        {
            var calc = new Calculator();
            Console.WriteLine(calc.Add(5, 3));

            // Using LINQ
            var numbers = new[] { 1, 2, 3, 4, 5 };
            var evens = numbers.Where(n => n % 2 == 0).ToList();

            // Pattern matching with switch
            foreach (var num in numbers)
            {
                var result = num switch
                {
                    1 => "one",
                    2 => "two",
                    _ => "other"
                };
                Console.WriteLine(result);
            }

            // Null-conditional and null-coalescing
            string? nullable = null;
            var length = nullable?.Length ?? 0;

            // Using record
            var person = new Person("Alice", 30);
            Console.WriteLine(person);
        }
    }
}
